oak-store-document/src/test/java/org/apache/jackrabbit/oak/plugins/document/MongoVersionGCSupportDefaultNoBranchTest.java

```diff
@@ -0,0 +1,431 @@
+        Stats(VersionGCStats stats, int nodesDeleteMany) {
+            if (stats == null) {
+                throw new IllegalArgumentException("stats must not be null");
+            }
+            if (nodesDeleteMany < 0) {
+                throw new IllegalArgumentException("nodesDeleteMany must be positive");
+            }
+            this.versionGCStats = stats;
+            this.nodesDeleteMany = nodesDeleteMany;
+        }
+    }
+
+    private static Predicate<NodeDocument> splitDocsWithClusterId(final int clusterId) {
+        return new Predicate<NodeDocument>() {
+            @Override
+            public boolean apply(@Nullable NodeDocument doc) {
+                if (!Utils.isPreviousDocId(doc.getId())) {
+                    return false;
+                }
+                Path p = doc.getPath();
+                p = p.getAncestor(1);
+                Revision rev = Revision.fromString(p.getName());
+                return rev.getClusterId() == clusterId;
+            }
+        };
+    }
+
+    private static final Set<NodeDocument.SplitDocType> GC_TYPES = EnumSet.of(
+            DEFAULT_LEAF, COMMIT_ROOT_ONLY, DEFAULT_NO_BRANCH);
+
+    class MongoVersionGCSupportAccessor extends MongoVersionGCSupport {
+
+        public MongoVersionGCSupportAccessor(MongoDocumentStore store) {
+            super(store);
+        }
+
+        protected Iterable<NodeDocument> identifyGarbage(Set<SplitDocType> gcTypes,
+                RevisionVector sweepRevs, long oldestRevTimeStamp) {
+            return super.identifyGarbage(gcTypes, sweepRevs, oldestRevTimeStamp);
+        }
+    }
+    // using AbstractTwoNodeTest as a helper rather than subclassing to simplify things
+    private AbstractTwoNodeTest helper;
+    private DocumentStoreFixture fixture;
+    private DocumentStore store1;
+    private DocumentStore store2;
+    protected DocumentNodeStore ds1;
+    protected DocumentNodeStore ds2;
+    private VersionGCSupport gcSupport1;
+    private List<String> ids = Lists.newArrayList();
+
+    private Clock clock;
+    private AtomicInteger offset = new AtomicInteger(0);
+
+    private final static AtomicInteger nodesDeleteMany = new AtomicInteger(0);
+
+    @Parameterized.Parameters(name="{0}")
+    public static java.util.Collection<DocumentStoreFixture> fixtures() {
+        List<DocumentStoreFixture> fixtures = Lists.newArrayList();
+        if (MONGO.isAvailable()) {
+            fixtures.add(new MongoFixture() {
+                @Override
+                public DocumentStore createDocumentStore(Builder builder) {
+                    try {
+                        MongoConnection connection = MongoUtils.getConnection();
+                        connections.add(connection);
+                        MongoDatabase db = connection.getDatabase();
+                        final AtomicReference<String> noEx = new AtomicReference<>(null);
+                        MongoTestDatabase testDb = new MongoTestDatabase(db, noEx, noEx, noEx) {
```

Comment:
Could this also go to `CountingMongoDatabase` and `CountingMongoCollection`?

Reply:
yes, can (and without interfering with existing tests). done now in https://github.com/apache/jackrabbit-oak/pull/651/commits/d82ba32226e09ec545e348bc57367167052bfdbb